src/next/main.ts

```diff
@@ -121,10 +122,10 @@ function report(eventName: string, properties?: Properties, measurements?: Measu
 }
 
 const ALLOWED_KEYS_FOR_ERROR: string[] = [
-    "errorCode",// Preserve 0 for no error, 1 for general error.
-    "name",     // For name of an Error.
-    "message",  // For message of an Error.
-    "stack",    // For callstack of an Error.
+    "errorCode",    // Preserve "0" for no error, "1" for general error.
+    "message",      // For message of an Error.
+    "stack",        // For callstack of an Error.
+    "isUserError",  // Indicator of user error. Possible values: "true", "false".

```

Comment:
You mentioned that boolean values are tricky for dimension entries. If that's still true, then something like `errorType` with the value `userError` or `systemError` makes more sense.

Reply:
If the opposite side of `userError` is exactly `systemError`, then what you comment here makes sense. Do you think there would be another kind of `nonSystemNonUserError` of which we should take care?